### ISSUE TITLE
OS X Support?

### ISSUE THREAD

**Issue by kbenzie** [8BitPimp/colib#1]
Are there any plans for OS X support?

I believe the asm could be shared with Linux given that the default compiler for OS X is Apple Clang which uses the same asm syntax, unless I'm mistaken.

**Comment by bit-hack** [comment]
If I ever have access to a mac device i'm more then happy to make sure it works there.  I think your right however and that porting should not be too much of a pain.  If apple have changed their ABI however (like microsoft did) then it would need a specific OS X version.

**Comment by kbenzie** [comment]
Maybe we can have a look at it over lunch sometime, I'll bring my MacBook.

**Comment by bit-hack** [comment]
I merged in most of your changes for adding OSX support.  However I didnt take the auto selection of the target in CMake as I have remember why I didn’t do that previously.  From what I can see CMake is really crappy when it comes to detecting the host processor, however I need to be able to select between x86, Arm, Mips, PowerPC, etc all running on Linux.  Its easy from C but seems hard to do in CMake.

I do need to test the osx stuff after the merge, as I may have broken stuff.

**Comment by kbenzie** [comment]
Just pulled the latest changes, four of the five tests pass but `test_fpregs` crashes, is there some form of debug output I can get from lldb for you?

**Comment by bit-hack** [comment]
This is in part because I buggered up my little test suite.  It would report a crash even if a test fails an assert. I am about to patch this shortly.  I think however this fail may be caused by coroutines not preserving the FPU state, but perhaps we can have a look at this on your mac some time?

**Comment by kbenzie** [comment]
Sure I can bring it on Thursday again if that suites you.

**Comment by bit-hack** [comment]
Yeah that would be cool!  I can actualy see the same problem in release mode with the windows x64 build.  I seems to be an sse instruction, movaps, dereferencing a nullptr, in my test coroutine.  It seems to happen in the prologue, so I suspect I have violated the ABI somehow :/
